### PR TITLE
Fixed a bad merge that resulted in breaking communication between AgentC...

### PR DESCRIPTION
**Pull request by amrhassan**
...ontroller and CSAgent.

### CODE REVIEW

File: lib/JumpScale/grid/serverbase/Daemon.py [hunk 1]
```diff
@@ -167,22 +167,6 @@ def processRPC(self, cmd, data, returnformat, session, category=""):
...
-            if not self.cmdsInterfaces.has_key(category):
-                return returnCodes.METHOD_NOT_FOUND, "", None
-
-            cmdinterface= self.cmdsInterfaces[category]
-            if hasattr(cmdinterface, cmd):
-                ffunction = getattr(cmdinterface, cmd)
-            else:
-                return returnCodes.METHOD_NOT_FOUND, "", None
-
-            self.cmds[cmdkey] = ffunction
```
Comment (grimpy) [inline lib/JumpScale/grid/serverbase/Daemon.py]: Why is this removed?

File: apps/agentcontroller/controller.py [hunk 2]
```diff
@@ -679,7 +679,7 @@ def listJobs(self, session=None):
             result.append(jobresult)
         return result
 
-    def getAllJumpscripts(self, bz2_compressed=True, types=('processmanager', 'jumpscripts'), session=None):
+    def getJumpscripts(self, bz2_compressed=True, types=('processmanager', 'jumpscripts'), session=None):
```
Comment (grimpy) [inline apps/agentcontroller/controller.py]: the method is just in the jsagent with this name if you need it changed change it there to

File: lib/JumpScale/grid/serverbase/Daemon.py (hunk 1 above)
Comment (amrhassan) [inline lib/JumpScale/grid/serverbase/Daemon.py]: Because it was factored out into `ffunction = self.command_handler(command_category=category, command=cmd)` then the old code was brought back again and now the two blocks of code do the same thing twice consecutively.

File: apps/agentcontroller/controller.py (hunk 2 above)
Comment (amrhassan) [inline apps/agentcontroller/controller.py]: @grimpy You meant the call in  `lib/JumpScale/grid/jumpscripts/JumpscriptFactory.py`?

Comment (grimpy) [inline apps/agentcontroller/controller.py]: yes, i see you changed it now

File: lib/JumpScale/grid/serverbase/Daemon.py (hunk 1 above)
Comment (grimpy) [inline lib/JumpScale/grid/serverbase/Daemon.py]: ok
